pipeline/matching/evaluation/analytics.py

```diff
@@ -0,0 +1,24 @@
+from typing import Dict, Iterator, List
+
+from pipeline.types import Match, MatchingInput, MatchingInternalAnalytics, User
+
+
+class CollectInternalAnalytics:
```

Comment:
How about we rename this to `MatchingMetricsCollector`? It is sometimes helpful for the class name to be a noun and the methods to be verbs.

You can rename the method to be `count_proposed_matches_per_user()` to match the verb format, or just leave it as-is.